src/PlusDataCollection/Atracsys/vtkPlusAtracsysTracker.cxx

```diff
@@ -0,0 +1,396 @@
+    LOG_ERROR("Failed to initialize Atracsys tool: DataSource Id is missing.");
+    continue;
+    }
+
+    TRACKING_TYPE toolTrackingType;
+    XML_READ_ENUM2_ATTRIBUTE_NONMEMBER_OPTIONAL(TrackingType, toolTrackingType, toolDataElement, "ACTIVE", ACTIVE, "PASSIVE", PASSIVE);
+    if (toolTrackingType == ACTIVE)
+    {
+      // active tool, can be loaded directly into FtkGeometryIdMappedToToolId
+      int ftkGeometryId = -1;
+      XML_READ_SCALAR_ATTRIBUTE_NONMEMBER_OPTIONAL(int, GeometryId, ftkGeometryId, toolDataElement);
+      if (ftkGeometryId != -1)
+      {
+        std::pair<int, std::string> thisTool(ftkGeometryId, toolId);
+        this->Internal->FtkGeometryIdMappedToToolId.insert(thisTool);
+      }
+      else
+      {
+        LOG_ERROR("Active tool with Id " << toolId << " is missing a GeometryId");
+        return PLUS_FAIL;
+      }
+    }
+    else if (toolTrackingType == PASSIVE)
+    {
+      // passive tool, load into PlusIdMappedToGeometryFilename to have geometry loaded in InternalConnect
+      const char* geometryFile;
+      if ((geometryFile = toolDataElement->GetAttribute("GeometryFile")) != NULL)
+      {
+        std::pair<std::string, std::string> thisTool(toolId, geometryFile);
+        this->Internal->PlusIdMappedToGeometryFilename.insert(thisTool);
+      }
+      else
+      {
+        LOG_ERROR("Passive tool with Id " << toolId << " is missing GeometryFile (.ini).");
+        return PLUS_FAIL;
+      }
+    }
+  }
+
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusAtracsysTracker::WriteConfiguration(vtkXMLDataElement* rootConfigElement)
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::WriteConfiguration");
+  XML_FIND_DEVICE_ELEMENT_REQUIRED_FOR_WRITING(deviceConfig, rootConfigElement);
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusAtracsysTracker::Probe()
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::Probe");
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusAtracsysTracker::InternalConnect()
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::InternalConnect");
+
+  ATRACSYS_RESULT result;
+
+  // Connect to tracker
+  result = this->Internal->Tracker->Connect();
+  if (result != ATR_SUCCESS && result != AtracsysTracker::ATRACSYS_RESULT::WARNING_CONNECTED_IN_USB2)
+  {
+    LOG_ERROR(this->Internal->Tracker->ResultToString(result));
+    return PLUS_FAIL;
+  }
+  else if (result == AtracsysTracker::ATRACSYS_RESULT::WARNING_CONNECTED_IN_USB2)
+  {
+    LOG_WARNING(this->Internal->Tracker->ResultToString(result));
+  }
+
+  // get device type
+  this->Internal->Tracker->GetDeviceType(this->Internal->DeviceType);
+
+  // if spryTrack, setup for onboard processing and disable extraneous marker info streaming
+  if (this->Internal->DeviceType == AtracsysTracker::DEVICE_TYPE::SPRYTRACK_180)
```

Comment:
Could this low-level configuration pushed down into the AtracsysTracker class?
It would be nice if users of AtracsysTracker class would not need to know model-specific settings.